### ISSUE TITLE
Segmentation fault while running mqtt_demo_mutual_auth example

### ISSUE THREAD

**Issue by surukonda** [aws/aws-iot-device-sdk-embedded-C#1718]
While running the mqtt_demo_mutual_auth demo example if network connectivity is lost, the program exits with a Segmentation fault error.

## Steps to Reproduce ##
1. Run the mqtt_demo_mutual_auth demo (while device is connected to wifi)
2. Switch-off the wifi router or wifi hotspot after few messages are successfully sent to the IoT Core
3. Program tries to re-connect and fail with Segmentation fault error

Probably a condition to check for valid connection before calling `Openssl_Disconnect( &networkContext );` at line 1580 may resolve this.

## Log ##

````bash
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
openssl connection staus 0
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1335] Creating an MQTT connection to abcdefghijklm-ats.iot.us-west-1.amazonaws.com.
[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt_serializer.c:970] CONNACK session present bit not set.
[INFO] [MQTT] [core_mqtt_serializer.c:912] Connection accepted.
[INFO] [MQTT] [core_mqtt.c:1563] Received MQTT CONNACK successfully from broker.
[INFO] [MQTT] [core_mqtt.c:1829] MQTT connection established with the broker.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1104] MQTT connection successfully established with broker.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1367] A clean MQTT connection is established. Cleaning up all the stored outgoing publishes.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1385] Subscribing to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1168] SUBSCRIBE sent for topic testclient/example/topic to broker.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=3.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:991] Subscribed to the topic testclient/example/topic. with maximum QoS 1.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 2.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 3.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 2.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 2.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 4.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 3.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 3.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 5.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 4.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 4.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 6.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 5.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 5.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1462] Unsubscribing from the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1211] UNSUBSCRIBE sent for topic testclient/example/topic to broker.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 6.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 6.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1001] Unsubscribed from the topic testclient/example/topic.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1486] Disconnecting the MQTT connection with abcdefghijklm-ats.iot.us-west-1.amazonaws.com.
[INFO] [MQTT] [core_mqtt.c:2149] Disconnected from the broker.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1578] Demo completed successfully.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1584] Short delay before starting the next iteration....

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
openssl connection staus 0
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1335] Creating an MQTT connection to abcdefghijklm-ats.iot.us-west-1.amazonaws.com.
[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt_serializer.c:970] CONNACK session present bit not set.
[INFO] [MQTT] [core_mqtt_serializer.c:912] Connection accepted.
[INFO] [MQTT] [core_mqtt.c:1563] Received MQTT CONNACK successfully from broker.
[INFO] [MQTT] [core_mqtt.c:1829] MQTT connection established with the broker.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1104] MQTT connection successfully established with broker.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1367] A clean MQTT connection is established. Cleaning up all the stored outgoing publishes.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1385] Subscribing to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1168] SUBSCRIBE sent for topic testclient/example/topic to broker.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=3.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:991] Subscribed to the topic testclient/example/topic. with maximum QoS 1.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 9.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 10.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 9.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 9.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=2.
[INFO] [MQTT] [core_mqtt.c:1162] Ack packet deserialized with result: MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1175] State record updated. New state=MQTTPublishDone.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1016] PUBACK received for packet id 10.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:731] Cleaned up outgoing publish packet with packet id 10.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 11.


[INFO] [MQTT] [core_mqtt.c:886] Packet received. ReceivedBytes=40.
[INFO] [MQTT] [core_mqtt.c:1045] De-serialized incoming PUBLISH packet: DeserializerResult=MQTTSuccess.
[INFO] [MQTT] [core_mqtt.c:1058] State record updated. New state=MQTTPubAckSend.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:820] Incoming QOS : 1.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:835] Incoming Publish Topic Name: testclient/example/topic matches subscribed topic.
Incoming Publish message Packet Id is 1.
Incoming Publish Message : Hello World!.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 12.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1429] Sending Publish to the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1266] PUBLISH sent for topic testclient/example/topic to broker with packet ID 13.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1450] Delay before continuing to next iteration.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1462] Unsubscribing from the MQTT topic testclient/example/topic.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1211] UNSUBSCRIBE sent for topic testclient/example/topic to broker.


[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1486] Disconnecting the MQTT connection with abcdefghijklm-ats.iot.us-west-1.amazonaws.com.
[INFO] [MQTT] [core_mqtt.c:2149] Disconnected from the broker.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1578] Demo completed successfully.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:1584] Short delay before starting the next iteration....

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
[ERROR] [Sockets] [sockets_posix.c:138] Failed to resolve DNS: Hostname=abcdefghijklm-ats.iot.us-west-1.amazonaws.com, ErrorCode=-2.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:679] Failed to establish a TLS connection.
backoffstatus 0, BackoffAlgorithmSuccess:0.........
[WARN] [DEMO] [mqtt_demo_mutual_auth.c:652] Connection to the broker failed. Retrying connection after 93 ms backoff.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
[ERROR] [Sockets] [sockets_posix.c:138] Failed to resolve DNS: Hostname=abcdefghijklm-ats.iot.us-west-1.amazonaws.com, ErrorCode=-2.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:679] Failed to establish a TLS connection.
backoffstatus 0, BackoffAlgorithmSuccess:0.........
[WARN] [DEMO] [mqtt_demo_mutual_auth.c:652] Connection to the broker failed. Retrying connection after 427 ms backoff.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
[ERROR] [Sockets] [sockets_posix.c:138] Failed to resolve DNS: Hostname=abcdefghijklm-ats.iot.us-west-1.amazonaws.com, ErrorCode=-2.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:679] Failed to establish a TLS connection.
backoffstatus 0, BackoffAlgorithmSuccess:0.........
[WARN] [DEMO] [mqtt_demo_mutual_auth.c:652] Connection to the broker failed. Retrying connection after 1028 ms backoff.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
[ERROR] [Sockets] [sockets_posix.c:138] Failed to resolve DNS: Hostname=abcdefghijklm-ats.iot.us-west-1.amazonaws.com, ErrorCode=-2.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:679] Failed to establish a TLS connection.
backoffstatus 0, BackoffAlgorithmSuccess:0.........
[WARN] [DEMO] [mqtt_demo_mutual_auth.c:652] Connection to the broker failed. Retrying connection after 494 ms backoff.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
[ERROR] [Sockets] [sockets_posix.c:138] Failed to resolve DNS: Hostname=abcdefghijklm-ats.iot.us-west-1.amazonaws.com, ErrorCode=-2.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:679] Failed to establish a TLS connection.
backoffstatus 0, BackoffAlgorithmSuccess:0.........
[WARN] [DEMO] [mqtt_demo_mutual_auth.c:652] Connection to the broker failed. Retrying connection after 4984 ms backoff.
[INFO] [DEMO] [mqtt_demo_mutual_auth.c:628] Establishing a TLS session to abcdefghijklm-ats.iot.us-west-1.amazonaws.com:8883.
[ERROR] [Sockets] [sockets_posix.c:138] Failed to resolve DNS: Hostname=abcdefghijklm-ats.iot.us-west-1.amazonaws.com, ErrorCode=-2.

[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:679] Failed to establish a TLS connection.
backoffstatus 1, BackoffAlgorithmSuccess:0.........
[ERROR] [DEMO] [mqtt_demo_mutual_auth.c:644] Connection to the broker failed, all attempts exhausted.
openssl connection staus 1
[ERROR] [DEMO] [mqtt_demo_mutual_auth.c:1567] Failed to connect to MQTT broker abcdefghijklm-ats.iot.us-west-1.amazonaws.com.
Segmentation fault
````

**Comment by hrishikesheb** [comment]
Hi @surukonda since you were able to build the mqtt_demo_mutual_auth can you help me how you did that
I have commented my error in this issue. Please have a look at this and suggest the possible solution.
https://github.com/aws/aws-iot-device-sdk-embedded-C/issues/1695#issuecomment-916667090

**Comment by hrishikesheb** [comment]
Hi @surukonda since you are able to build the mqtt_demo_mutual_auth demo can you help me how you did that. I have successfully built the demo but it does not include mqtt_demo_mutual_auth or any other mqtt demos.
![image](https://user-images.githubusercontent.com/70017670/132981490-9299a348-048e-4f04-9341-b3dab273cb2a.png)

Can you help me with this

**Comment by surukonda** [comment]
> Hi @surukonda since you are able to build the mqtt_demo_mutual_auth demo can you help me how you did that. I have successfully built the demo but it does not include mqtt_demo_mutual_auth or any other mqtt demos.
> ![image](https://user-images.githubusercontent.com/70017670/132981490-9299a348-048e-4f04-9341-b3dab273cb2a.png)
> 
> Can you help me with this

Hi @hrishikesheb - you can find the demo [here](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/main/demos/mqtt/mqtt_demo_mutual_auth)

**Comment by hrishikesheb** [comment]
Thanks for replying @surukonda now i am successfully able to run the demo (after setting all the macros) but I get this error. Please have a look at it and suggest possible solution.

[INFO] [DEMO] [mqtt_demo_mutual_auth.c:645] Establishing a TLS session to <my_aws-endpoint>-ats.iot.us-west-2.amazonaws.com:8883.
[ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:817] Failed to receive data over network: SSL_read failed: ErrorStatus=EVP lib.
[ERROR] [MQTT] [core_mqtt_serializer.c:2402] A single byte was not read from the transport: transportStatus=-1.
[ERROR] [MQTT] [core_mqtt.c:1570] CONNACK recv failed with status = MQTTRecvFailed.
[ERROR] [MQTT] [core_mqtt.c:1841] MQTT connection failed with status = MQTTRecvFailed.
[ERROR] [DEMO] [mqtt_demo_mutual_auth.c:1126] Connection with MQTT broker failed with status MQTTRecvFailed.
[WARN] [DEMO] [mqtt_demo_mutual_auth.c:683] Connection to the broker failed. Retrying connection after 455 ms backoff.

**Comment by surukonda** [comment]
> Thanks for replying @surukonda now i am successfully able to run the demo (after setting all the macros) but I get this error. Please have a look at it and suggest possible solution.
> 
> [INFO] [DEMO] [mqtt_demo_mutual_auth.c:645] Establishing a TLS session to <my_aws-endpoint>-ats.iot.us-west-2.amazonaws.com:8883.
> [ERROR] [Transport_OpenSSL_Sockets] [openssl_posix.c:817] Failed to receive data over network: SSL_read failed: ErrorStatus=EVP lib.
> [ERROR] [MQTT] [core_mqtt_serializer.c:2402] A single byte was not read from the transport: transportStatus=-1.
> [ERROR] [MQTT] [core_mqtt.c:1570] CONNACK recv failed with status = MQTTRecvFailed.
> [ERROR] [MQTT] [core_mqtt.c:1841] MQTT connection failed with status = MQTTRecvFailed.
> [ERROR] [DEMO] [mqtt_demo_mutual_auth.c:1126] Connection with MQTT broker failed with status MQTTRecvFailed.
> [WARN] [DEMO] [mqtt_demo_mutual_auth.c:683] Connection to the broker failed. Retrying connection after 455 ms backoff.
> 
> Also can you tell me the steps you took to successfully run the demo without any errors.

Request you to go through the steps specified [here](https://github.com/aws/aws-iot-device-sdk-embedded-C#configuring-mutual-authentication-demos-of-mqtt-and-http)

**Comment by archigup** [comment]
Thanks for the bug report, looking into the segfault 👍

**Comment by archigup** [comment]
@surukonda Please open a new issue, as to not detract from this one.

**Comment by archigup** [comment]
@hrishikesheb  Please open a new issue, as to not detract from this one.
Edit: Sorry pinged wrong person.

**Comment by muneebahmed10** [comment]
Hi @surukonda,

> Probably a condition to check for valid connection before calling `Openssl_Disconnect( &networkContext );` at line 1580 may resolve this

That looks to be the cause of your errors. I've created #1720 to fix this. Thank you for bringing this to our attention.
